### DISCUSSION TITLE
"Wrong or expired session token"

### DISCUSSION THREAD

**Discussion by nkarabulut** [Mathieu2301/TradingView-API#77] [Show and tell]
Hello,

I need session id cookie ..

if (!process.argv[2]) throw Error('Please specify your \'sessionid\' cookie');

How and where can I get the token?

can you explain briefly ?

I try get token to logged browsers developer inspector's cookie tab key sessionid.
this string like af6w4h1svh2sgreqsa53sukyf16ayrdaq 

I use in client like this

const client = new TradingView.Client({
  token: "af6w4h1svh2sgreqsa53sukyf16ayrdaq",
});

but it says token expired . Browser is logged and have session.

what's wrong. 

I need token for private indicators. 

Thanks.

**Comment by Mathieu2301** [comment]
Weird... Can you try this with your token?
```ru
node examples/CustomTimeframe.js xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
For me the token is 32 characters in size.
Did you try to remove all cookies or `Clear site data` (in the `Storage` tab) ?

I will investigate...

Thank you for sponsoring me :)

**Reply by nkarabulut** [reply]
I try 

node examples/CustomTimeframe.js xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

and its working. Thanks.

You are welcome. I think it is a project that should be supported.

**Comment by Mathieu2301** [comment]
I just added a new `loginUser(...)` function.
You have an example [here](https://github.com/Mathieu2301/TradingView-API/blob/main/examples/UserLogin.js) !

**Reply by Mathieu2301** [reply]
When you use this function, you can view the session in the Session History section of your account settings.
<img width="378" alt="image" src="https://user-images.githubusercontent.com/21021423/155845999-db12c488-9b16-4a26-b29d-b65a1b90332d.png">
For security reasons, I suggest you to `Log Out` all old sessions that you no longer use.

**Reply by nkarabulut** [reply]
Login support was very necessary .Thank you.

**Comment by nkarabulut** [comment]
I partially did what I needed but .

now I need a second-based chart, and finally, if I can do that, it will be finished.
src/types.js
inside

/**
  * @typedef {'1' | '3' | '5' | '15' | '30'
  * | '45' | '60' | '120' | '180' | '240'
  * | '1D' | '1W' | '1M' | 'D' | 'W' | 'M'} TimeFrame
  */

I added the lines

/**
  * @typedef {'1s' | '3s' | '5s' | '15s' | '30s'
* | '1' | '3' | '5' | '15' | '30'
* | '45' | '60' | '120' | '180' | '240'
  * | '1D' | '1W' | '1M' | 'D' | 'W' | 'M'} TimeFrame
  */

But,I couldn't solve this error (I'm trying 30 seconds)

Critical error: invalid parameters method: create_series. args: "[$prices, s1, ser_1, 30h, 100]"
Client critical error: [
   '{unknown_session_id,<<"create_study">>,<<"cs_u8fC1HbT5nn2">>,\n' ...
   
Thanks for your help by the way.

**Reply by Mathieu2301** [reply]
> Critical error: invalid parameters method: create_series. args: "[$prices, s1, ser_1, 30h, 100]"

I see `30h` 🤔🤔

JSDoc typedef is only for intellisense and I didn't put timeframes which are only available with premium because there are too many. Technically you can set '26s', '4s', '58s', etc... So you don't need to edit the JSDoc.

Does the `CustomTimeframe.js` example work for you?

**Comment by nkarabulut** [comment]
Thanks for reply I try CustomTimeframe.js and I fixed code its works.
